test/e2e/main/main_test.go

```diff
@@ -1450,3 +1452,81 @@ func createCacheWithCR(cacheName string, nameSpace string, clusterName string) *
 		},
 	}
 }
+
+func TestPodDegradationAfterOOM(t *testing.T) {
+	t.Parallel()
+
+	//Creating Infinispan cluster
+	ispn := tutils.DefaultSpec(testKube)
+	name := strcase.ToKebab(t.Name())
+	ispn.Name = name
+	ispn.Labels = map[string]string{"test-name": t.Name()}
+	ispn.Spec.Replicas = 2
+	ispn.Spec.Container.Memory = "256Mi"
+
+	testKube.CreateInfinispan(ispn, tutils.Namespace)
+	defer testKube.CleanNamespaceAndLogOnPanic(tutils.Namespace, ispn.Labels)
+	testKube.WaitForInfinispanPods(int(ispn.Spec.Replicas), tutils.SinglePodTimeout, ispn.Name, tutils.Namespace)
+	testKube.WaitForInfinispanCondition(ispn.Name, ispn.Namespace, ispnv1.ConditionWellFormed)
+
+	//Creating cache
+	cacheName := "failover-cache"
+	template := `<infinispan><cache-container><replicated-cache name ="` + cacheName +
+		`"><encoding media-type="text/plain"/></replicated-cache></cache-container></infinispan>`
+	veryLongValue := GenerateStringWithCharset(6000)
+	hostAddr, client := tutils.HTTPClientAndHost(ispn, testKube)
+	createCacheWithXMLTemplate(cacheName, hostAddr, template, client)
+
+	//Generate tons of random entries
+	for key := 1; key < 50000; key++ {
+		keyURL := fmt.Sprintf("%v/%v", cacheURL(cacheName, hostAddr), key)
+		headers := map[string]string{"Content-Type": "text/plain"}
+		_, error := client.Post(keyURL, veryLongValue, headers)
+
+		if error != nil {
+			fmt.Printf("\nERROR for key=%d, Description=%s\n", key, error)
+			break
+		}
+	}
+
+	//Check if all pods are running and they are not degraded
+	testKube.WaitForInfinispanPods(int(ispn.Spec.Replicas), tutils.SinglePodTimeout, ispn.Name, tutils.Namespace)
+	testKube.WaitForInfinispanCondition(ispn.Name, ispn.Namespace, ispnv1.ConditionWellFormed)
+
+	//Verify whether the pod restarted for an OOM exception
+	hasOOMhappened := false
```

Comment:
Shouldn't this check occur before we call `WaitForInfinispanPods` and `WaitForInfinispanCondition`?

Also, I would expect that the test should fail if `!hasOOMhappened`, because it's not possible to test pod recovery from OOM if we can't confirm it has indeed occurred.

Reply:
I am moving the check before "WaitForInfinispanPods" and "WaitForInfinispanCondition", I don't get the second part though.
The condition checks whether "kube.AreAllPodsReady(podList) && hasOOMhappened", so I am excepting that if !hasOOMhappened the test panic. Am I losing something else?

Reply:
I am assuming that with that memory and with that values it will always go in OOM (it breaks with only 2500 occurrences of 50000). Whether I don't find "hasOOMhappened" to true the test is going to fail